### PR TITLE
client: cleanup RPCClient options

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

use optional security argument so we don't have to pass empty security config every time.

### CODE REVIEW

File: internal/client/client_fail_test.go [hunk 1]
```diff
@@ -59,9 +59,8 @@ func TestPanicInRecvLoop(t *testing.T) {
 	defer server.Stop()
 
 	addr := fmt.Sprintf("%s:%d", "127.0.0.1", port)
-	rpcClient := NewRPCClient(config.Security{}, func(c *RPCClient) {
-		c.dialTimeout = time.Second / 3
-	})
+	rpcClient := NewRPCClient()
+	rpcClient.dialTimeout = time.Second / 3
```
Comment (sticnarf) [inline internal/client/client_fail_test.go:63]: I'm a bit curious about the `opts` design. If the options can be set by this way, why need we pass opts in `NewRPCClient`?

Comment (disksing) [reply internal/client/client_fail_test.go:63]: opts can be used by library users that do not have access to private fields. For `dialTimeout`, we only need to update it in the same package (at least for now), so don't have to use opts.

Comment (sticnarf) [reply internal/client/client_fail_test.go:63]: Ah, I see. If any field that needs to be modified by the library user, we provide a function returning `Opt` like `WithSecurity`.

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**